changelog.d/21673_add_http_headers_to_opentelemetry_source.feature.md

```diff
@@ -0,0 +1,3 @@
+The `opentelemetry` source can now be configured to enrich log events with the source IP and any HTTP headers received in the OTLP/HTTP request.
```

Comment:
Let's also mention that now a `host_key` can be specified. 

Note, I am skeptical about the connection of the `host_key` with `source_ip`. I understand there's precedence in other sources but I can see an argument in favor of making it a different setting.

Reply:
Commit ea5c56ff4d87da0eddcac88e63ecde17d5570503 mentioned this, but I reverted that commit to reduce the scope of this PR, since you said you were not sure that `host_key` should mean `source_ip`.